### ISSUE TITLE
Metadata Artist ReTag strips 'Feat.' or '&' and Secondary Artists

### ISSUE THREAD

**Issue by Alan111S** [lidarr/Lidarr#1580]
**Is your feature request related to a problem? Please describe.**
When retagging files, the Artist (as in Contributing Artist) is stored with only the Main Artist. Any Secondary Artists or Featured Artists are removed. e.g. 'SexyBack' is now tagged with Artist 'Justin Timbalake' instead of 'Justin Timberlake (feat. Timbaland)'
![Retag Before](https://user-images.githubusercontent.com/17717393/92856554-e8d66f00-f436-11ea-9b95-df0a17e09dd5.PNG)
![Retag After](https://user-images.githubusercontent.com/17717393/92856559-ea079c00-f436-11ea-8f8d-778ba55fd73a.PNG)

**Describe the solution you'd like**
I'd like Retag to keep the proper Artist List from MusicBrainz. Also, would be useful to add a new token {TrackArtist} to use with the Rename function. As you can see in the attached the filename has the Artist replaced with Various Artists.

**Comment by Alan111S** [comment]
Suppose this might be useful:- 
![MusicBrainz](https://user-images.githubusercontent.com/17717393/92857772-88483180-f438-11ea-8900-ca9297e142fd.PNG)

**Comment by Chuchodavids** [comment]
Having this same issue with the metadata tagger. This is driving me crazy.

**Comment by elwayman02** [comment]
Seeing the same issue as well. I really wanted to use Lidarr to clean up my tags and filenames, but dropping the featured artists from my tracks is a big regression in my metadata.

![image](https://user-images.githubusercontent.com/1325482/118071590-eed17680-b35c-11eb-8ae0-15f9661818a5.png)

**Comment by FinalDoom** [comment]
Still present in current Lidarr. Pretty annoying TBH. One of these days I'll look at fixing it, but no promises.

**Comment by Kolbasz12** [comment]
I wish we could take some of the "smarts" from Beets and implement it into Lidarr for the way it handles tagging.

1. feat artists
2. continuous track numbers vs disc folders.

**Comment by FinalDoom** [comment]
Probably worth a separate issue/feature request for porting over beets logic. This one's pretty specific.

On that separate idea, my thoughts:
I wonder if a less work-intensive solution could be something like:
* Add a configuration item for script or webhook to kick off
* Instructions for people to set up beets or something else.. docker container.. ??

My current workflow just has music dumped in a folder for review/fixing through Picard before ultimate library import. I'll have to check out beets though. 

Maybe since beets is opensource though enough logic could be ported from it to be worthwhile.

**Comment by grasshide** [comment]
+1 for this issue. 
Lidarr's feat removal of feat artists from songtitles makes it impossible for me to find songs related to an artist in any search.

**Comment by Coder84619** [comment]
+1 any update?

**Comment by pozineg** [comment]
...same problem with featuring artists metadata tags missing. Shame it hasn't been addressed for such a long time. Can you talented guys look into it, please.

**Comment by bakerboy448** [comment]
The primary development team is engaged in over five applications, including Lidarr, Prowlarr, Radarr, Readarr, Whisparr, and Sonarr. Currently, there is approximately one-ish active developer contributing to these projects, with none of the contributors working on them full-time. Additionally, all members involved, whether in development or support roles, are volunteers and do not receive compensation for their contributions. Each team member balances their commitments alongside full-time jobs, family responsibilities, and other personal obligations.

As a result, the list of tasks to be completed across these projects, along with the various backends and modules, is extensive, while the available time for addressing these tasks is limited.

**Comment by thekiefs** [comment]
This is similar to https://github.com/lidarr/Lidarr/issues/681, where we needed diverged rename logic based on the type of album. Perhaps it makes sense to implement something like MusicBrainz, where they have a scripting language meant to cover all rename possibilities.

<img width="992" alt="Image" src="https://github.com/user-attachments/assets/25aab6e1-ea3b-4df3-9f96-560ed3195ee2" />

**Comment by andrewtwelch** [comment]
Part of implementing this would involve a change to the metadata API to provide the artist per track (at least where it differs). I know there was talk about eventually making that code public, is there a rough idea on how long until that happens, or any other options to get those changes made prior to that happening?

Ideally I'd set this up so that it can fall back to the album artist if the track artist is missing, but we'd need to make sure that providing an artist per track is possible (even just where it differs from the artist if that's easier) from the metadata API before starting on this.
